core/src/main/java/uk/gov/justice/services/core/dispatcher/Dispatcher.java

```diff
@@ -64,6 +78,11 @@ void register(final Object handler) {
         handlerRegistry.register(handler);
     }
 
+    private JsonEnvelope doDispatch(final JsonEnvelope envelope, final boolean isSynchronous) {
+        checkAccessControl(envelope);
+        return (JsonEnvelope) getMethod(envelope, isSynchronous).execute(envelope);
```

Comment:
I think we should have the "if" statement here, so we're not using the exception to control flow.